tests/test_repository.py

```diff
@@ -21,7 +21,21 @@
 from twine import utils
 
 
+@pytest.fixture()
+def repo():
+
```

Comment:
Please align this with the style in the rest of the project. There's no need for the whitespace around this or to bind it to `repo` prior to returning it

Reply:
Can this fixture accept `request` and inspect the `params` provided for a `repository_url` param and do the right thing?

Reply:
> Please align this with the style in the rest of the project. There's no need for the whitespace around this or to bind it to `repo` prior to returning it

So something like this?

```
@pytest.fixture()
def repo():
    return repository.Repository(
        repository_url=utils.DEFAULT_REPOSITORY,
        username="username",
        password="password",
    )
```

Reply:
> Can this fixture accept `request` and inspect the `params` provided for a `repository_url` param and do the right thing?

Sorry I don't follow. What are you referring to when you say `request` and `params` ?

Reply:
Yes

Reply:
Fixture whitespace removed in latest commit

Reply:
> Can this fixture accept `request` and inspect the `params` provided for a `repository_url` param and do the right thing?

Are you talking about something like this?

https://github.com/pypa/twine/blob/2c30b1e9c7936a6edd2e54c44852b58461fbc771/tests/test_wheel.py#L24-L32

From what I understand, this is in a way to parametrize the tests using this fixture, with different values provided in `request.param`, and in our case, using it for `repository_url`.

So do the change entail adding more repositories e.g. `"https://test.pypi.org/legacy/"` ?

I was also thinking if this can also help at https://github.com/pypa/twine/pull/604/files#diff-ad6a1d777d14851a581d57579ad2299eR168

where I am updating the fixture with a new url in the test myself 

`repo.url = "https://bad.repo.com/legacy"`

and using request.param will help in avoiding that?

Reply:
@deveshks I think you've got the idea. @sigmavirus24 what params did you have in mind? Just `utils.TEST_REPOSITORY`?

FWIW, I'm a little wary of parameterized fixtures, because that means your tests have to be aware of the parameterization, but it might not be obvious to the reader of the tests.

Reply:
So I think a parametrized fixture would be fine for all the URLs we're parametrizing elsewhere, especially when `repository_url` as a test parameter isn't being used except to construct the repository. However, to @bhrutledge's point, yes that particular test shouldn't use the fixture.

Reply:
In that case, I would not parametrise this fixture.